cmake/common.cmake

```diff
@@ -1,660 +1,681 @@
-    endif()
-endmacro()
-
-macro(spl_add_compile_options pattern)
-    message(DEBUG "spl_add_compile_options: pattern=${pattern}")
-    cmake_parse_arguments(ADD_SOURCE_ARGS "" "" "COMPILE_OPTIONS" ${ARGN})
-    message(DEBUG "spl_add_source: ADD_SOURCE_ARGS_COMPILE_OPTIONS=${ADD_SOURCE_ARGS_COMPILE_OPTIONS}")
-
-    file(GLOB_RECURSE files ${CMAKE_CURRENT_LIST_DIR}/${pattern})
-    message(DEBUG "spl_add_compile_options: files=${files}")
-
-    if(files)
-        foreach(file ${files})
-            set_source_files_properties(${file} PROPERTIES COMPILE_OPTIONS "${ADD_SOURCE_ARGS_COMPILE_OPTIONS}")
-        endforeach()
-    endif()
-endmacro()
-
-macro(spl_add_include includeDirectory)
-    _spl_get_absolute_path(to_be_appended ${includeDirectory})
-    list(APPEND INCLUDES ${to_be_appended})
-endmacro()
-
-macro(spl_add_test_source fileName)
-    _spl_get_absolute_path(to_be_appended ${fileName})
-    list(APPEND TEST_SOURCES ${to_be_appended})
-endmacro()
-
-macro(_spl_get_google_test)
-    # GoogleTest requires at least C++14
-    set(CMAKE_CXX_STANDARD 14)
-
-    # Make source of googletest configurable
-    if(NOT DEFINED SPL_GTEST_URL)
-        set(SPL_GTEST_URL https://github.com/google/googletest.git)
-    endif(NOT DEFINED SPL_GTEST_URL)
-
-    if(NOT DEFINED SPL_GTEST_TAG)
-        set(SPL_GTEST_TAG v1.14.0)
-    endif(NOT DEFINED SPL_GTEST_TAG)
-
-    include(FetchContent)
-    FetchContent_Declare(
-        googletest
-        GIT_REPOSITORY ${SPL_GTEST_URL}
-        GIT_TAG ${SPL_GTEST_TAG}
-    )
-
-    # Prevent overriding the parent project's compiler/linker settings on Windows
-    if(WIN32)
-        set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
-    endif()
-
-    # The Python version we want to use is in the PATH, so disable any search for it.
-    # Got it from here: https://stackoverflow.com/questions/73514630/make-cmake-not-search-for-python-components-on-reconfigure
-    set(CMAKE_DISABLE_FIND_PACKAGE_Python TRUE)
-
-    FetchContent_MakeAvailable(googletest)
-    include(GoogleTest)
-
-    enable_testing()
-endmacro(_spl_get_google_test)
-
-macro(spl_create_component)
-    cmake_parse_arguments(CREATE_COMPONENT "" "LONG_NAME;LIBRARY_TYPE" "" ${ARGN})
-    # Set the default library type to OBJECT if not provided
-    if(NOT CREATE_COMPONENT_LIBRARY_TYPE)
-        set(CREATE_COMPONENT_LIBRARY_TYPE "OBJECT")
-    endif()
-
-    file(RELATIVE_PATH component_path ${CMAKE_SOURCE_DIR} ${CMAKE_CURRENT_LIST_DIR})
-    _spl_slash_to_underscore(component_name ${component_path})
-
-    # Add debug options based on build configuration (kit)
-    if(BUILD_KIT STREQUAL test)
-        # Exclude all components from the ALL target.
-        # Components which have no tests shall not be automatically compiled when ALL target is executed.
-        add_library(${component_name} EXCLUDE_FROM_ALL OBJECT ${SOURCES})
-        target_compile_options(${component_name} PRIVATE ${VARIANT_ADDITIONAL_COMPILE_C_FLAGS} -ggdb --coverage)

```

Comment:
If you remove this "feature" than you have to mark it as a breaking change.
I would have not mixed the two things, fixing inline and getting rid of the variant additional c flags.